pkg/controller/operators/decorators/operator.go

```diff
@@ -356,9 +372,23 @@ func (c *Component) Reference() (ref *operatorsv1.RichReference, err error) {
 		ObjectReference: truncated,
 	}
 
-	out, _ := jsonpath.Read(c.UnstructuredContent(), componentConditionsJSONPath)
-	if out == nil {
-		return
+	query := componentConditionsJQ
+	switch c.GroupVersionKind() {
+	case csvGVK:
+		query = csvConditionsJQ
+	}
+	iter := query.Run(c.UnstructuredContent())
+
+	var out interface{}
+	for {
+		v, ok := iter.Next()
```

Comment:
if the query returns no results, so `ok` is false here and we break -- is there an issue passing a blank `interface{}` into the `Decode` call below? Should we check it's length?